src/Splunk.OpenTelemetry.AutoInstrumentation/ServiceNameWarning.cs

```diff
@@ -29,11 +29,11 @@ internal static class ServiceNameWarning
     private const char AttributeListSplitter = ',';
     private const char AttributeKeyValueSplitter = '=';
 
-    private static bool _warningEmitted;
+    private static int _warningEmitted;
 
     internal static void SendOnMissingServiceName()
     {
-        if (_warningEmitted)
+        if (Interlocked.CompareExchange(ref _warningEmitted, 1, 0) == 1)
```

Comment:
```suggestion
        if (Interlocked.CompareExchange(ref _warningEmitted, 1, 0) != 0)
```

(premature optimization 😆 )